examples/plot_05_train_test_split.py

```diff
@@ -0,0 +1,180 @@
+#
+# .. code-block:: python
+#
+#   X_train, X_test, y_train, y_test = sklearn_train_test_split(
+#       X=X, y=y, test_size=0.2, random_state=0)
+#
+# would return:
+#
+# .. code-block:: python
+#
+#   TypeError: got an unexpected keyword argument 'X'
+#
+# In general, in Python, positional arguments are useful to prevent typos such as:
+
+# %%
+X_train, X_test, y_train, y_test = sklearn_train_test_split(
+    y, X, test_size=0.2, random_state=0
+)
+
+# %%
+# where ``y`` and ``X`` are inverted in the arguments and no warning is returned.
+# This is where skore comes in handy.
+
+# %%
+# Train-test split in skore
+# =========================
+
+# %%
+# skore has its own :func:`skore.train_test_split` that wraps scikit-learn's
+# :func:`sklearn.model_selection.train_test_split`.
+
+# %%
+# Expliciting the positional arguments for ``X`` and ``y``
+# --------------------------------------------------------
+
+# %%
+# First of all, naturally, it can be used as a simple drop-in replacement for
+# scikit-learn:
+
+# %%
+X_train, X_test, y_train, y_test = skore.train_test_split(
+    X, y, test_size=0.2, random_state=0
+)
+print(X_train, y_train)
+print(X_test, y_test)
+
+# %%
+# .. note::
+#
+#   The outputs of :func:`skore.train_test_split` are intentionally exactly the same as
+#   :func:`sklearn.model_selection.train_test_split`, so the user can just use the
+#   skore version as a drop-in replacement of scikit-learn.
+
+# %%
+# Contrary to scikit-learn, skore allows users to explicit the``X`` and ``y``, making
+# detection of eventual issues easier:
+
+# %%
+X_train, X_test, y_train, y_test = skore.train_test_split(
+    X=X, y=y, test_size=0.2, random_state=0
+)
+print(X_train, y_train)
+print(X_test, y_test)
+
+# %%
+# Moreover, when passing ``X`` and ``y`` explicitly, the ``X``'s are always returned
+# before the ``y``'s, even when they are inverted:
+
+# %%
+arr = np.arange(10).reshape((5, 2))
+arr_train, arr_test, X_train, X_test, y_train, y_test = skore.train_test_split(
+    arr, y=y, X=X, test_size=0.2, random_state=0
+)
+print(X_train, y_train)
+print(X_test, y_test)
+
+# %%
+# Automatic diagnostics: raising methodological warnings
+# ------------------------------------------------------
+#
+# In machine learning, class-imbalance (the classes in a dataset are not equally
+# represented) requires a specific modelling.
+# For example, in a dataset with 95% majority class and 5% minority class, a model that
+# always predicts the majority class will have a 95% accuracy, while it would be
+# useless for identifying the minority class.
+# Hence, it is important to detect when we have class-imbalance.
+#
+# Suppose that we have imbalanced data :
+
+# %%
+X = [[1]] * 4
+y = [0, 1, 1, 1]
+
+# %%
+# In that case, :func:`skore.train_test_split` raises a warning telling the user that
+# there is class imbalance:
+
+# %%
+X_train, X_test, y_train, y_test = skore.train_test_split(
+    X=X, y=y, test_size=0.2, random_state=0
+)
```

Comment:
note: indeed, the warning is displayed quite badly... we need to take care of this at some point.